OPHD/States/Planet.cpp

```diff
@@ -66,3 +68,118 @@ void Planet::update()
 	const auto spriteFrameOffset = NAS2D::Point{mTick % 8 * PlanetSize.x, ((mTick % 64) / 8) * PlanetSize.y};
 	renderer.drawSubImage(mImage, mPosition, NAS2D::Rectangle<int>::Create(spriteFrameOffset, PlanetSize));
 }
+
+namespace
+{
+	Planet::Attributes parsePlanet(const NAS2D::Xml::XmlElement* xmlNode);
+
+	void parseElementValue(Planet::PlanetType& destination, const NAS2D::Xml::XmlElement* element);
+	void parseElementValue(Planet::Hostility& destination, const NAS2D::Xml::XmlElement* element);
+
+	const std::unordered_map<std::string, Planet::PlanetType> planetTypeTable
+	{
+		{"None", Planet::PlanetType::None},
+		{"Mercury", Planet::PlanetType::Mercury},
+		{"Mars", Planet::PlanetType::Mars},
+		{"Ganymede", Planet::PlanetType::Ganymede}
+	};
+
+	const std::unordered_map<std::string, Planet::Hostility> hostilityTable
+	{
+		{"None", Planet::Hostility::None},
+		{"Low", Planet::Hostility::Low},
+		{"Medium", Planet::Hostility::Medium},
+		{"High", Planet::Hostility::High}
+	};
+
+
+	void parseElementValue(Planet::PlanetType& destination, const NAS2D::Xml::XmlElement* element)
+	{
+		destination = stringToEnum(planetTypeTable, element->getText());
+	}
+
+
+	void parseElementValue(Planet::Hostility& destination, const NAS2D::Xml::XmlElement* element)
+	{
+		destination = stringToEnum(hostilityTable, element->getText());
+	}
+}
+
+
+std::vector<Planet::Attributes> parsePlanetAttributes()
+{
+	const std::string rootElementName("Planets");
+	auto xmlDocument = openXmlFile("planets/PlanetAttributes.xml", rootElementName);
+
+	std::vector<Planet::Attributes> planetAttributes;
+
+	auto rootElement = xmlDocument.firstChildElement(rootElementName);
+	for (const auto* node = rootElement->iterateChildren(nullptr);
+		node != nullptr;
+		node = rootElement->iterateChildren(node))
+	{
+		std::string elementName("Planet");
+		if (node->value() != elementName)
+		{
+			throw std::runtime_error(xmlDocument.value() + " missing " + elementName + " tag");
+		}
+		planetAttributes.push_back(parsePlanet(node->toElement()));
+	}
+
+	return planetAttributes;
+}
+
+
+namespace 
+{
+	Planet::Attributes parsePlanet(const NAS2D::Xml::XmlElement* xmlNode)
+	{
```

Comment:
Not that it matters, though you could potentially have put this in the first unnamed namespace, possibly even replacing the function prototype. Though there is nothing wrong about re-opening an unnamed namespace and keeping the implementation separate.